src/v/crypto/crypto.cc

```diff
@@ -40,4 +44,39 @@ std::ostream& operator<<(std::ostream& os, format_type type) {
         return os << "DER";
     }
 }
+
+namespace internal {
+EVP_MD* get_md(digest_type type) {
+    // Map of pre-fetched MD pointers.  This replaces the older way of getting
+    // a digest pointer via something like `EVP_sha256()`.  The old way is
+    // slower compared to pre-fetching the algorithm.
+    static thread_local absl::node_hash_map<digest_type, EVP_MD_ptr> md_map;
+    auto it = md_map.find(type);
+    if (it == md_map.end()) {
+        auto alg = fmt::to_string(type);
+        auto md_ptr = EVP_MD_fetch(nullptr, alg.c_str(), "?provider=fips");
+        if (!md_ptr) {
+            throw ossl_error(fmt::format("Failed to fetch algorithm {}", alg));
+        }
+        [[maybe_unused]] auto res = md_map.insert_or_assign(
+          type, EVP_MD_ptr(md_ptr));
+        vassert(
+          res.second, "Failed to insert/create the fetched algorithm {}", alg);
```

Comment:
this is fine. just fyi, you don't need [[maybe_unused]] if you use vassert because vassert is always compiled in so res is always used. but with vanilla assert it is compiled out in Release builds so res is not used and you need the annotation.

Reply:
doh!